### PR TITLE
Add first draft of installation guide

### PR DESCRIPTION
**Pull request by lbianchi-lbl**
## Fixes/Addresses:
 
Last remaining item of #113 (which apparently got converted to #145 when I clicked on it?)

## Summary/Motivation

This PR contains the first draft for the installation guide for ProteusLib. The current structure is the following:

- *Introduction/Using Conda environments*
  - Covers what is a Conda environment, why it is the recommended way to install ProteusLib, how to install Conda
- *General installation*
  - Covers how to install `proteuslib` in the general/default case of read-only, or *user-mode* installation, i.e. without needing to clone the ProteusLib repository
- *For ProteusLib developers*
  - Covers how to install `proteuslib` in *developer mode*, i.e. from a clone of the ProteusLib repo that will then be used to contribute back to the ProteusLib codebase

### Motivation/thoughts for reviewers

**TL;DR/Summary: A good installation guide is hard to write well and (partially) project-specific, so feel free to be honest in the review regarding both form and substance in this guide**

- Installation guides are in my experience very tricky to get right, as there is a number of prerequisites to install and/or configure before the user is able to install the actual software
- Following the Diataxis nomenclature for docs, they're arguably closest to a Tutorial in their goals (especially installation guides in a "Getting started" section)
- On the other hand, I find that the sheer number of unrelated steps and components involved can be confusing for the user if they're provided without any context of what they're used for
  - See e.g. a (perfectly valid) question of "do I need to use Conda/virtual environments to install ProteusLib?"
- Precisely because it's tricky, we thought the most reasonable thing to do would be to have a draft of the installation guide to get the conversation started (which is useable, meaning that the steps in it should work), and then iterate during the review process
- **Therefore, I encourage everyone to comment and provide feedback, questions, or corrections to this, and to consider any of the choices described there up for discussion**
- The basic structure (i.e. what's currently in the PR) is informed by our experience and general practises for Python projects, but every project has slightly different requirements and "developer audience", and it makes sense that the general installation procedures to be adapted to match that

## Changes proposed in this PR:
- Add installation guide to the docs as dedicated page in the "Getting Started" section

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by lbianchi-lbl** [comment]
A reminder: at any point, you can click on the link provided by the ReadTheDocs check to view a built version of the docs including the changes being proposed in this PR: https://proteuslib--144.org.readthedocs.build/en/144/

File: docs/getting_started/install.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+	.. code-block:: shell
+
+		git clone https://github.com/nawi-hub/proteuslib && cd proteuslib
+
+#. Install ProteusLib and the development dependencies using ``pip`` and the ``requirements-dev.txt`` file:
+
+	.. code-block:: shell
+
+		pip install -r requirements-dev.txt
```
Comment (bknueven) [inline docs/getting_started/install.rst:87]: Shouldn't we also tell developers to use the `-e` flag?

**Review by bknueven — APPROVED** [review]

File: docs/getting_started/install.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+Introduction
+------------
+
+.. _about-conda:
+
+Using Conda environments
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Conda environments are a way to create and manage multiple sets of packages and/or Python versions on the same system without incurring into conflicts.
```
Comment (adam-a-a) [inline docs/getting_started/install.rst:12]:
```suggestion
Conda environments are a way to create and manage multiple sets of packages and/or Python versions on the same system without incurring conflicts.
```

File: docs/getting_started/install.rst (hunk 1 above)
Comment (lbianchi-lbl) [reply docs/getting_started/install.rst:87]: That's a good point. The `-e/--editable` flag is, by design, built into `requirements-dev.txt`, with the underlying idea that `pip install -r requirements-dev.txt` is the single, consistent command to run to set up the developer environment.

It could be worth to note this here. Perhaps using a `.. note::` directive pointing to the (currently empty) developer how-to guide?

File: docs/getting_started/install.rst (hunk 2 above)
Comment (lbianchi-lbl) [reply docs/getting_started/install.rst:12]: Good catch, thanks!

**Review by adam-a-a — APPROVED** [review]
Thanks for the thorough explanation! I think this draft does the job nicely.

File: docs/how_to_guides/how_to_contribute_development.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.. _developer-guide:
+
+How to contribute to ProteusLib's development
```
Comment (adam-a-a) [inline docs/how_to_guides/how_to_contribute_development.rst:3]: By the way, should we add an issue for this how-to to keep it in mind?

File: docs/getting_started/install.rst (hunk 1 above)
Comment (bknueven) [reply docs/getting_started/install.rst:87]: I had forgotten (and didn't check) that it was built in to `requirements-dev.txt`. I'm happy with this as-it-is then.

**Review by bknueven — DISMISSED** [review]

**Comment by lbianchi-lbl** [comment]
@adam-a-a @bknueven we've just realized with @ksbeattie that for the "General installation" it's not possible to use `pytest` since that's a developer dependency (and it's not clear whether the unit tests themselves would be installed when doing `pip install` not in editable mode). So we changed the last step in the "General installation" to something that doesn't require pytest.

Let me know if what we have now makes sense, or if there's a better way (keeping in mind that this is just as a "smoke test", i.e. to have some way of detecting early if something went wrong, rather than a comprehensive check).

**Review by ksbeattie — APPROVED** [review]
LGTM

**Review by adam-a-a — APPROVED** [review]
